### PR TITLE
Move rjsf form submission to xhr for better error visibility

### PR DESCRIPTION
**Pull request by jbalboni**
Connects to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/3168
Connects to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/3144

Using xhr instead of fetch gives us some more error visibility by being able to see timeout and abort errors. This may not help at all, but we'll at least be consistent across browsers. It should address the issue with fetch and Edge 14, too.

### CODE REVIEW

File: src/js/common/schemaform/actions.js [hunk 1]
```diff
@@ -59,46 +58,86 @@ export function submitForm(formConfig, form) {
...
+          const results = JSON.parse(responseBody);
+          resolve(results);
+        } else {
+          const error = new Error(`vets_server_error: ${req.statusText}`);
+          Raven.captureException(error, {
+            extra: {
+              clientError: false,
+              statusText: req.statusText
+            }
+          });
```
Comment (U-DON) [inline src/js/common/schemaform/actions.js]: This gets duplicated quite a bit. Might it be helpful to have this as a function within the scope of this promise?

**Review by U-DON — APPROVED** [review]
LGTM.

**Review by cehsu — APPROVED** [review]
Looks good to me.

File: src/js/common/schemaform/actions.js (hunk 1 above)
Comment (jbalboni) [inline src/js/common/schemaform/actions.js]: I ended up moving this out more and calling it from the promise's `catch` function. It's possible the way I had it before would miss errors in the promise callback function.

**Pull request merged**